Examples/package-info/Sources/package-info/main.swift

```diff
@@ -13,22 +14,16 @@ let packagePath = AbsolutePath(#file).parentDirectory.parentDirectory.parentDire
 
 // There are several levels of information available.
 // Each takes longer to load than the level above it, but provides more detail.
-let diagnostics = DiagnosticsEngine(handlers: [{ print($0)}])
+
+let observability = ObservabilitySystem({ print("\($0): \($1)") })
+
 let workspace = try Workspace(forRootPackage: packagePath)
-let manifest = try tsc_await { workspace.loadRootManifest(at: packagePath, diagnostics: diagnostics, completion: $0) }
-guard !diagnostics.hasErrors else {
-    fatalError("error loading manifest: \(diagnostics)")
-}
-
-let package = try tsc_await { workspace.loadRootPackage(at: packagePath, diagnostics: diagnostics, completion: $0) }
-guard !diagnostics.hasErrors else {
-    fatalError("error loading package: \(diagnostics)")
-}
-
-let graph = try workspace.loadPackageGraph(rootPath: packagePath, diagnostics: diagnostics)
-guard !diagnostics.hasErrors else {
-    fatalError("error loading graph: \(diagnostics)")
-}
+
+let manifest = try tsc_await { workspace.loadRootManifest(at: packagePath, observabilityScope: observability.topScope, completion: $0) }
```

Comment:
Do we lose the context here for where the diagnostic is being emitted from?

Reply:
not sure i follow, in what way do we loose context?

Reply:
I just meant that the previous error messages said  "error loading manifest” etc.  It looks like that code has been removed though, so do they just say “error” now?  Or is the 'error loading manifest” what the `$0` is in the observability handler?

I was simply noticing the removal of a string containing context visible to the user.

Reply:
so this specific code is an example of how to use the workspace apis, and not part of SwiftPM itself

the example used to check for "hasError" on the diagnostics, for cases where a diagnostics was emitted, but that check is not required and actually an anti-pattern. if a critical error was found the code would exit on the throw, otherwise diagnostics will be printed via the handler configured at the top of the example

Reply:
@abertelrud note this specific change is part of https://github.com/apple/swift-package-manager/pull/3777 which this PR builds on top, so probably worth reviewing and merging https://github.com/apple/swift-package-manager/pull/3777 first, then I will rebase this PR on top which will make the diff much smaller

Reply:
Thanks for pointing that out.  Yes, I noticed that after leaving this comment.  I’ll review that one first.